sympy/solvers/polysys.py

```diff
@@ -185,6 +194,16 @@ def solve_generic(polys, opt):
     .. [Cox97] D. Cox, J. Little, D. O'Shea, Ideals, Varieties
     and Algorithms, Springer, Second Edition, 1997, pp. 112
 
+    Raises
+    ========
+
+    NotImplementedError
+        The system is not zero-dimensional (does not have a finite number of solutions)
+
+        ``strict`` is True and at least one polynomial in the Groebner basis does not
+        have all its solutions expressible in radicals (after it is converted to a
+        univariate polynomial)
```

Comment:
I think that a new error class would be better than `NotImplementedError` here. Something like `UnsolvableFactorError`.

Reply:
I have added a new error class as you suggest.